ftw/globalstatusmessage/browser/controlpanel.py

```diff
@@ -10,5 +33,82 @@ class StatusMessageEditForm(controlpanel.RegistryEditForm):
     description = _(u'Settings for Global Status Message.')
 
 
+class PublishingStatusMessageEditForm(StatusMessageEditForm):
+    """
+    A special edit form which handles the publishing of the global status message
+    to a "ftw.publisher.receiver" instance.
+
+    Please note that the buttons are rendered in the order as they are defined.
+    """
+    @button.buttonAndHandler(plone_messagefactory(u"Save"), name='save')
+    def handleSave(self, action):
+        # Redeclare the save button because it is not inherited from the parent class.
+        super(PublishingStatusMessageEditForm, self).handleSave(self, action)
+
+    @button.buttonAndHandler(_(u"Save and publish"), name='save_and_publish')
+    def handleSaveAndPublish(self, action):
+        """
+        The additional button to send to a "ftw.publisher.receiver" instance.
+        """
+        self.handleSave(self, action)
+
+        self.send_to_receiver()
+
+        # Consume Plone messages set by `handleSave` so we can set our own message.
+        self.context.cookies[STATUSMESSAGEKEY] = None
+        self.context.response.expireCookie(STATUSMESSAGEKEY, path='/')
+        annotations = IAnnotations(self.context)
+        annotations[STATUSMESSAGEKEY] = None
```

Comment:
Unfortunately, the messages are not cleared because of the `wrong` HTTP status code: https://github.com/plone/Products.statusmessages/blob/499f14b5bd578d88751643652aae645f32e1dfe6/Products/statusmessages/adapter.py#L64